summon.lua

```diff
@@ -0,0 +1,195 @@
+	["magma_cube"] = mtMagmaCube,
+	["mooshroom"] = mtMooshroom,
+	["ocelot"] = mtOcelot,
+	["pig"] = mtPig,
+	["rabbit"] = mtRabbit,
+	["sheep"] = mtSheep,
+	["silverfish"] = mtSilverfish,
+	["skeleton"] = mtSkeleton,
+	["slime"] = mtSlime,
+	["snowman"] = mtSnowGolem,
+	["spider"] = mtSpider,
+	["squid"] = mtSquid,
+	["villager"] = mtVillager,
+	["witch"] = mtWitch,
+	["wither"] = mtWither,
+	["wolf"] = mtWolf,
+	["zombie"] = mtZombie,
+	["zombie_pigman"] = mtZombiePigman,
+
+	-- 1.10 and below
+	["Bat"] = mtBat,
+	["Blaze"] = mtBlaze,
+	["CaveSpider"] = mtCaveSpider,
+	["Chicken"] = mtChicken,
+	["Cow"] = mtCow,
+	["Creeper"] = mtCreeper,
+	["EnderDragon"] = mtEnderDragon,
+	["Enderman"] = mtEnderman,
+	["Ghast"] = mtGhast,
+	["Giant"] = mtGiant,
+	["Guardian"] = mtGuardian,
+	["Horse"] = mtHorse,
+	["LavaSlime"] = mtMagmaCube,
+	["MushroomCow"] = mtMooshroom,
+	["Ozelot"] = mtOcelot,
+	["Pig"] = mtPig,
+	["Rabbit"] = mtRabbit,
+	["Sheep"] = mtSheep,
+	["Silverfish"] = mtSilverfish,
+	["Skeleton"] = mtSkeleton,
+	["Slime"] = mtSlime,
+	["SnowMan"] = mtSnowGolem,
+	["Spider"] = mtSpider,
+	["Squid"] = mtSquid,
+	["Villager"] = mtVillager,
+	["VillagerGolem"] = mtIronGolem,
+	["Witch"] = mtWitch,
+	["Wither"] = mtWither,
+	["Wolf"] = mtWolf,
+	["Zombie"] = mtZombie,
+	["PigZombie"] = mtZombiePigman
+}
+
+local Projectiles =
+{
+	["arrow"] = cProjectileEntity.pkArrow,
+	["egg"] = cProjectileEntity.pkEgg,
+	["ender_pearl"] = cProjectileEntity.pkEnderPearl,
+	["fireworks_rocket"] = cProjectileEntity.pkFirework,
+	["fishing_float"] = cProjectileEntity.pkFishingFloat,
+	["fireball"] = cProjectileEntity.pkGhastFireball,
+	["potion"] = cProjectileEntity.pkSplashPotion,
+	["small_fireball"] = cProjectileEntity.pkFireCharge,
+	["snowball"] = cProjectileEntity.pkSnowball,
+	["wither_skull"] = cProjectileEntity.pkWitherSkull,
+	["xp_bottle"] = cProjectileEntity.pkExpBottle,
+
+	-- 1.10 and below
+	["Arrow"] = cProjectileEntity.pkArrow,
+	["Fireball"] = cProjectileEntity.pkGhastFireball,
+	["FireworksRocketEntity"] = cProjectileEntity.pkFirework,
+	["FishingFloat"] = cProjectileEntity.pkFishingFloat,
+	["SmallFireball"] = cProjectileEntity.pkFireCharge,
+	["Snowball"] = cProjectileEntity.pkSnowball,
+	["ThrownEgg"] = cProjectileEntity.pkEgg,
+	["ThrownEnderpearl"] = cProjectileEntity.pkEnderPearl,
+	["ThrownExpBottle"] = cProjectileEntity.pkExpBottle,
+	["ThrownPotion"] = cProjectileEntity.pkSplashPotion,
+	["WitherSkull"] = cProjectileEntity.pkWitherSkull
+}
+
+function HandleSummonCommand(Split, Player)
+	if Split[2] == nil then
+		Player:SendMessageInfo("Usage: " .. Split[1] .. " <entityname> [x] [y] [z]")
+	else
+		local X = Player:GetPosX()
+		local Y = Player:GetPosY()
+		local Z = Player:GetPosZ()
+		local World = Player:GetWorld()
+
+		if Split[3] ~= nil then
+			X = tonumber(Split[3])
+			local RelativeX = loadstring(Split[3]:gsub("~", "return " .. Player:GetPosX() .. "+0"))
+			if RelativeX then
+				X = select(2, pcall(setfenv(RelativeX, {})))
```

Comment:
This could be refactored into a separate function, `EvaluateCoord` that takes the current player coord and the string from the split, and returns a single number.

Does Vanilla support expressions such as `~ * 2`? Because this parser won't evaluate them properly.